infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -117,14 +118,16 @@ public class StreamLogFiles implements StreamLog {
      */
     private final ReadWriteLock resetLock = new ReentrantReadWriteLock();
 
+    private final FileSystemAgent fsAgent;
+
     /**
      * Returns a file-based stream log object.
      *
      * @param serverContext Context object that provides server state such as epoch,
      *                      segment and start address
      * @param noVerify      Disable checksum if true
      */
-    public StreamLogFiles(ServerContext serverContext, boolean noVerify) {
+    public StreamLogFiles(ServerContext serverContext, boolean noVerify, BatchProcessorContext batchProcessorContext) {
 
```

Comment:
I think the abstraction can be improved here, instead of having `StreanLogFiles` depend on `BatchProcessorContext`, you can move the `FileSystemAgent` to the `LogUnit` that way there is no circular dependency.